extension/src/plots/paths/collect.ts

```diff
@@ -118,6 +109,46 @@ const collectPathRevisions = (data: PlotsData, path: string): Set<string> => {
   return revisions
 }
 
+const collectPlotPathType = (
+  plotPath: PlotPath,
+  data: PlotsData,
+  hasChildren: boolean,
+  path: string
+) => {
+  if (hasChildren) {
+    return
+  }
+
+  const type = getType(data, path)
+
+  if (type) {
+    plotPath.type = type
+  }
+}
+
+const updateExistingPlotPath = (
+  acc: PlotPath[],
+  data: PlotsData,
+  hasChildren: boolean,
+  revisions: Set<string>,
+  path: string
+) =>
+  acc.map(existing => {
+    const plotPath = { ...existing }
+
+    if (existing.path !== path) {
+      return plotPath
+    }
+
+    plotPath.revisions = new Set([...existing.revisions, ...revisions])
+
+    if (!plotPath.type) {
```

Comment:
Not sure how common this bug is but when adding the plots to dvc.yaml in example-get-started and running an experiment, `plot diff` will have plots with `file not found` errors. This causes us to build our plot paths without types that never get added since the code only updates revisions if the path doesn't exist. 

I updated it to check the existing path's type existence but there might be a better way to handle this.

Reply:
I think we can remove the `!plotPath.type` check.